cpp/src/Ice/Exception.cpp

```diff
@@ -1,524 +1,5 @@
-        {
-            free(backtraceStrings);
-        }
-
-#endif
-        return stackTrace;
-    }
-}
-
-// TODO: make_shared is not noexcept.
-Ice::Exception::Exception(const char* file, int line) noexcept
-    : _file(file),
-      _line(line),
-      _what(nullptr),
-      _stackFrames(make_shared<vector<void*>>(getStackFrames()))
-{
-}
-
-Ice::Exception::Exception(const char* file, int line, string message)
-    : _file(file),
-      _line(line),
-      _whatString(make_shared<string>(std::move(message))),
-      _what(_whatString->c_str()),
-      _stackFrames(make_shared<vector<void*>>(getStackFrames()))
-{
-}
-
-const char*
-Ice::Exception::what() const noexcept
-{
-    return _what ? _what : ice_id();
-}
-
-const char*
-Ice::Exception::ice_file() const noexcept
-{
-    return _file;
-}
-
-int
-Ice::Exception::ice_line() const noexcept
-{
-    return _line;
-}
-
-string
-Ice::Exception::ice_stackTrace() const
-{
-    return getStackTrace(*_stackFrames);
-}
-
-void
-Ice::Exception::ice_enableStackTraceCollection()
-{
-    lock_guard lock(globalMutex);
-#if defined(ICE_DBGHELP)
-    if (process)
-    {
-        // Already initialized, just refresh.
-        if (!SymRefreshModuleList(process))
-        {
-            // TODO: SymRefreshModuleList occasionally fails with error code 3221225476; we retry once in this case.
-            // Note that calling GetLastError() does not reset the last error.
-            if (GetLastError() != 3221225476 || !SymRefreshModuleList(process))
-            {
-                throw std::runtime_error{
-                    "SymRefreshModuleList failed with " + IceInternal::errorToString(GetLastError())};
-            }
-        }
-    }
-    else
-    {
-        HANDLE currentProcess = GetCurrentProcess();
-        // duplicate handle as per https://learn.microsoft.com/en-us/windows/win32/debug/initializing-the-symbol-handler
-        if (!DuplicateHandle(currentProcess, currentProcess, currentProcess, &process, 0, FALSE, DUPLICATE_SAME_ACCESS))
-        {
-            throw std::runtime_error{
-                "DuplicateHandle on current process failed with " + IceInternal::errorToString(GetLastError())};
-        }
-
-        SymSetOptions(SYMOPT_LOAD_LINES | SYMOPT_DEFERRED_LOADS | SYMOPT_EXACT_SYMBOLS | SYMOPT_UNDNAME);
-        if (!SymInitialize(process, nullptr, TRUE))
-        {
-            DWORD initializeError = GetLastError();
-            CloseHandle(process);
-            process = nullptr;
-            throw std::runtime_error{"SymInitialize failed with " + IceInternal::errorToString(initializeError)};
-        }
-    }
-#elif defined(ICE_LIBBACKTRACE)
-    if (!bstate)
-    {
-        // Leaked, as libbacktrace does not provide an API to free this state.
-        bstate = backtrace_create_state(0, 1, ignoreErrorCallback, 0);
-    }
-#elif defined(ICE_BACKTRACE)
-    backTraceEnabled = true;
-#endif
-}
-
-ostream&
-Ice::operator<<(ostream& out, const Ice::Exception& ex)
```

Comment:
This PR eliminates the distinction `ice_print` / `operator<<`, which made this code too complicated.

As of this PR, (virtual) ice_print and `operator<<` are the same.